paasta_tools/secret_providers/vault.py

```diff
@@ -0,0 +1,45 @@
+import os
+
+from vault_tools.client.jsonsecret import get_plaintext
```

Comment:
Don't you have to gate this some how, as this i

Reply:
I could add a `try` `except`. We actually import this by calling `__import__` directly and the name is configurable. Maybe I should set the default to the base provider then if anyone tries to use secrets they'll get a NotImplemented exception. And at yelp we can set the provider to vault explicitly?